### PR TITLE
[10.0] backport sequence_reset_period from v11 to v10

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by alexis-via** [comment]
@etobella I backported your module sequence_reset_period to v10. Thanks for this very useful module!

**Review by etobella — APPROVED** [review]
Great!! 😉 
It is useful in some cases 😄

File: sequence_reset_period/models/ir_sequence.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def _create_date_range_seq(self, date):
+        self.ensure_one()
+        if not self.range_reset:
+            return super(IrSequence, self)._create_date_range_seq(date)
+        date_from, date_to = self._compute_date_from_to(date)
+        date_range = self.env['ir.sequence.date_range'].search(
+            [('sequence_id', '=', self.id), ('date_from', '>=', date),
+             ('date_from', '<=', date_to)], order='date_from desc', limit=1)
+        if date_range:
+            date_to = datetime.strptime(date_range.date_from,
```
Comment (lmignon) [inline sequence_reset_period/models/ir_sequence.py:44]: fields.Date.from_string

File: sequence_reset_period/models/ir_sequence.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                        '%Y-%m-%d') + timedelta(days=-1)
+            date_to = date_to.strftime('%Y-%m-%d')
```
Comment (lmignon) [inline sequence_reset_period/models/ir_sequence.py:46]: fields.Date.to_string

File: sequence_reset_period/models/ir_sequence.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            date_from = date_from - timedelta(days=date_from.weekday())
+            date_to = date_from + timedelta(days=6)
+        elif self.range_reset == 'monthly':
+            date_from = datetime_date(date_from.year, date_from.month, 1)
+            date_to = date_from + relativedelta(months=1)
+            date_to += relativedelta(days=-1)
+        elif self.range_reset == 'yearly':
+            date_from = datetime_date(date_from.year, 1, 1)
+            date_to = datetime_date(date_from.year, 12, 31)
+        return date_from.strftime('%Y-%m-%d'), date_to.strftime('%Y-%m-%d')
```
Comment (lmignon) [inline sequence_reset_period/models/ir_sequence.py:33]: fields.Date.to_string

**Review by lmignon — CHANGES_REQUESTED** [review]
@alexis-via Thank you for the backport. One comment: IMO we must use methods `from_string` and `to_string`defined on fields.Date to parse and format date value.

**Pull request closed without merging**